std/src/main/scala/algebra/std/map.scala

```diff
@@ -26,33 +50,80 @@ class MapEq[K, V](implicit ev: Eq[V], a: AdditiveMonoid[V]) extends Eq[Map[K, V]
     check(x, y) && check(y, x)
```

Comment:
why not check if the sizes are equal. If not, not equal. If they are equal we only have to `check(x, y)`